numba_dpex/ocl/oclimpl.py

```diff
@@ -282,27 +225,29 @@ def native_atomic_add(context, builder, sig, args):
     return builder.call(fn, fn_args)
 
 
+def support_atomic(dtype: types.Type) -> bool:
+    # This check should be the same as described in sycl documentation:
+    # https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#sec:atomic-references
+    # If atomic is not supported, it will be emulated by the sycl compiler.
+    return (
```

Comment:
Fine for now. My follow up changes will make it possible to use `dpctl.SyclDevice.has_aspect_atomic64` to refine the check.